src/transformers/modeling_utils.py

```diff
@@ -3714,12 +3384,19 @@ def _fix_key(key):
                     target_dtype = torch.float32
 
                 if param.device == torch.device("meta"):
-                    if not (is_quantized):
-                        set_module_tensor_to_device(model, key, "cpu", torch.empty(*param.size(), dtype=target_dtype))
+                    value = torch.empty(*param.size(), dtype=target_dtype)
+                    if getattr(
+                        quantizer, "requires_parameters_quantization", False
+                    ) or not quantizer.check_quantized_param(model, param_value=value, param_name=key, state_dict={}):
+                        set_module_tensor_to_device(model, key, "cpu", value)
                     else:
-                        set_module_quantized_tensor_to_device(
-                            model, key, "cpu", torch.empty(*param.size(), dtype=target_dtype)
-                        )
+                        raise NotImplementedError("Loading quantized parameters outside base model is not supported.")
+                        # TODO: consider whether to support this (e.g. quantized model head).
+                        # may need to create such model with quantized head for testing.
```

Comment:
This is one of the places not covered by tests, so initially I made these branches throw error.
Now to be consistent with the code behavior before,  I used `quantizer.create_quantized_param()`. Is it correct to leave "cpu" here (as it was before), or shall the device be determined from the "device_map"?
`set_module_quantized_tensor_to_device()` was not a valid choice since I got rid of it in `_load_state_dict_into_meta_model()` already.